code/modules/mob/holder.dm

```diff
@@ -43,58 +60,220 @@
 	if(!holder_type || buckled || pinned.len)
 		return
 
+	if ((grabber.hand == 0 && grabber.r_hand) || (grabber.hand == 1 && grabber.l_hand))//Checking if the hand is full
+		grabber << "Your hand is full!"
+		return
+
+
 	var/obj/item/weapon/holder/H = new holder_type(loc)
 	src.loc = H
 	H.name = loc.name
 	H.attack_hand(grabber)
+	H.contained = src
 
 	grabber << "You scoop up [src]."
 	src << "[grabber] scoops you up."
 	grabber.status_flags |= PASSEMOTES
 	return
 
 //Mob specific holders.
+//w_class mainly determines whether they can fit in trashbags. <=2 can, >=3 cannot
 /obj/item/weapon/holder/diona
 	name = "diona nymph"
-	desc = "It's a tiny plant critter."
+	desc = "It's a little plant critter."
 	icon_state = "nymph"
 	origin_tech = "magnets=3;biotech=5"
 	slot_flags = SLOT_HEAD | SLOT_OCLOTHING
+	w_class = 2
+
+/obj/item/weapon/holder/diona/dead
```

Comment:
Are the dead classes necessary?

The only things they seem to change are: `icon_state`, `desc`, and other minor vars. The same differences could be reached with if/else checks in the procs that initialize the holder object (if mob.stat == dead, change these vars), and in the mob's own death() proc (if is in holder, change it to this stuff).

Helper procs and vars would probably make it easier. Give each holder a desc_dead, an icon_state_dead, and whatever else vars need to be modified, and make a helper proc something like, `/obj/item/weapon/holder/proc/deadify()` which overwrites the vars as necessary. If a specific holder has some special shit attached to it being dead, then you can simply modify the childed `deadify()` proc.